checks/containers/sarus/check_gpu_ids.py

```diff
@@ -1,52 +1,37 @@
-import os
-import sys
-
 import reframe as rfm
 import reframe.utility.sanity as sn
 
 
-sys.path.append(os.path.dirname(os.path.abspath(__file__)))
-
-
-class BaseCheck(rfm.RunOnlyRegressionTest):
-    valid_systems = []
+@rfm.simple_test
+class SarusNvidiaSmiCheck(rfm.RunOnlyRegressionTest):
+    valid_systems = ['+nvgpu +sarus']
     valid_prog_environs = ['builtin']
-    sourcesdir = None
-    modules = ['sarus']
     num_tasks = 1
     num_tasks_per_node = 1
-    num_gpus_per_node = 1
-    sarus_image = 'ethcscs/cudasamples:8.0'
-    maintainers = ['amadonna', 'taliaga']
+    container_platform = 'Sarus'
+    image = 'nvidia/cuda:11.8.0-base-ubuntu18.04'
     tags = {'production'}
 
     @run_after('setup')
-    def setup_prerun(self):
-        self.prerun_cmds = ['sarus --version',
-                            f'sarus pull {self.sarus_image}']
-
-    @sanity_function
-    def assert_sanity(self):
-        return sn.assert_found(r'CHECK SUCCESSFUL', self.stdout)
+    def setup_gpu_options(self):
```

Comment:
1/ Is setting CUDA_VISIBLE_DEVICES needed for this specific test (nvidia-smi) ? 
2/ would it make sense to have a script inside the config dir instead of having to do that for each test ?

Reply:
1/ nvidia-smi does not check CUDA_VISIBLE_DEVICES

Reply:
I don't know if it would make a difference for nvidia-smi. But regarding 2, yes we can have a set of hooks or base test classes that define these hooks. @ekouts what do you think?

Reply:
It's needed for sarus to trigger the nvidia hook